routes/bridges/gassistant/index.js

```diff
@@ -0,0 +1,118 @@
+// -*- mode: js; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+//
+// This file is part of ThingEngine
+//
+// Copyright 2017-2019 The Board of Trustees of the Leland Stanford Junior University
+//
+// Author: Swee Kiat Lim <sweekiat@cs.stanford.edu>
+//
+// See COPYING for details
+"use strict";
+
+const Q = require('q');
+const express = require('express');
+
+const EngineManager = require('../../../almond/enginemanagerclient');
+const { actionssdk, Image } = require('actions-on-google');
+
+const userUtils = require('../../../util/user');
+
+var router = express.Router();
+
+class GoogleAssistantDelegate {
+    constructor(locale) {
+        this._buffer = '';
+        this._locale = locale;
+    }
+
+    send(text, icon) {
+        this._buffer += text + '\n';
+    }
+
+    sendPicture(url, icon) {
+        this._image = new Image({
+            url: url,
+            alt: url,
+        });
+    }
+
+    sendRDL(rdl, icon) {
+        this._buffer += rdl.displayTitle + '\n';
+    }
+
+    sendChoice(idx, what, title, text) {
+        this._buffer += title + '\n';
+    }
+
+    sendButton(title, json) {
+        // FIXME
+    }
+
+    sendLink(title, url) {
+        if (url === '/user/register') {
+            this._card = {
+                type: 'LinkAccount'
+            };
+        }
+        // FIXME handle other URL types
+    }
+
+    sendResult(message, icon) {
+        this._buffer += message.toLocaleString(this._locale) + '\n';
+    }
+
+    sendAskSpecial(what) {
+        this._askSpecial = what;
+    }
+}
+GoogleAssistantDelegate.prototype.$rpcMethods = ['send', 'sendPicture', 'sendChoice', 'sendLink', 'sendButton', 'sendAskSpecial', 'sendRDL', 'sendResult'];
+
+const app = actionssdk();
+
+// Register handler for Actions SDK
+
+app.intent('actions.intent.MAIN', (conv) => {
+    conv.ask('Hello! I\'m Almond, your virtual assistant.');
```

Comment:
Translation support here?
Also, by convention, user-visible strings use double quotes, mostly to avoid the escape on apostrophes...

(To mark a string as translatable, you usually wrap it in `req._("...")`. You don't have `req` here though. I don't know how to thread `_` through)

Reply:
Also, to most users I think we should show the Almond first-time welcome.
To do so, call `getOrOpenConversation` with a new ID, and that will start a new Almond conversation, which in anonymous mode will show the welcome screen.

Reply:
The welcome screen has the four sample replies ("Get a #cat gif" etc.) There's similar functionality in Google Assistant, called Suggestions, but with a limit of 25 characters max. So only "Get a #cat gif" will be complete while the rest will be cut off e.g. "Show me the weather for S" instead of "Show me the weather for San Francisco".
Should I just show the cat gif one instead?

Reply:
Yeah I think showing just the cat one will be ok.